baseSampleApp/src/main/java/org/kinecosystem/baseSampleApp/SampleBaseActivity.java

```diff
@@ -1,37 +1,44 @@
-package org.kinecosystem.sampleReceiverApp;
+package org.kinecosystem.baseSampleApp;
+
 
 import android.os.Bundle;
 import android.support.annotation.Nullable;
 import android.support.v7.app.AppCompatActivity;
 import android.widget.TextView;
 
-import org.kinecosystem.sampleReceiverApp.sampleWallet.OnBoarding;
-import org.kinecosystem.sampleReceiverApp.sampleWallet.SampleWallet;
+import org.kinecosystem.baseSampleApp.sampleWallet.OnBoarding;
+import org.kinecosystem.baseSampleApp.sampleWallet.SampleWallet;
 
 import kin.sdk.Balance;
 import kin.utils.Request;
 import kin.utils.ResultCallback;
 
-public class ReceiverMainActivity extends AppCompatActivity {
+public abstract class SampleBaseActivity extends AppCompatActivity {
     private boolean activityCreated = false;
 
+    protected abstract SampleWallet getSampleWallet();
+
+    protected int getLayoutId() {
+        return R.layout.sample_base;
+    }
+
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_main);
+        setContentView(getLayoutId());
 
         activityCreated = true;
 
         initAccountAndViews();
+        initCustomViews();
     }
 
-    private SampleWallet getSampleWallet() {
-        return ((ReceiverApplication) getApplicationContext()).getSampleWallet();
+    protected void initCustomViews() {
```

Comment:
not clear what is the difference between cusotom views and init account and view
maybe 2 methods 
init accounts
initviews

Reply:
The purpose was to allow the extending class to add more views. Decided to remove the method and simply extend onCreate 😄